templates/clusterclass-dev-test.yaml

```diff
@@ -0,0 +1,157 @@
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: ClusterClass
+metadata:
+  name: dev-test
+spec:
+  controlPlane:
+    ref:
+      apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+      kind: KubeadmControlPlaneTemplate
+      name: ${CLUSTER_NAME}-control-plane-template
+    machineInfrastructure:
+      ref:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1alpha7
+        kind: OpenStackMachineTemplate
+        name: ${CLUSTER_NAME}-control-plane-machine-template
+  infrastructure:
+    ref:
+      apiVersion: infrastructure.cluster.x-k8s.io/v1alpha7
+      kind: OpenStackClusterTemplate
+      name: ${CLUSTER_NAME}-openstackcluster-template
+  workers:
+    machineDeployments:
+      - class: default-worker
+        template:
+          bootstrap:
+            ref:
+              apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+              kind: KubeadmConfigTemplate
+              name: ${CLUSTER_NAME}-default-worker-bootstraptemplate
+          infrastructure:
+            ref:
+              apiVersion: infrastructure.cluster.x-k8s.io/v1alpha7
+              kind: OpenStackMachineTemplate
+              name: ${CLUSTER_NAME}-default-worker-machine-template
+  patches:
+  - name: controlPlaneImage
+    description: "Sets the OpenStack image that is used for creating the servers."
+    definitions:
+    - selector:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1alpha7
+        kind: OpenStackMachineTemplate
+        matchResources:
+          controlPlane: true
+      jsonPatches:
+      - op: add
+        path: /spec/template/spec/image
+        valueFrom:
+          template: |
```

Comment:
it is now an ImageFilter, https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/api/v1alpha8/types.go#L25-L32
so it'll be

```
name: ubuntu-2204-kube-{{ .builtin.controlPlane.version }}
```

Reply:
Are you guys fine without using patching? @EmilienM @lentzi90 , I have a comment a bit down about me trying it without any progress in `v1alpha8`.